src/Storefront/Framework/Cache/CacheWarmer/CacheWarmer.php

```diff
@@ -31,6 +34,11 @@ public function warmUp(?string $cacheId = null): void
         $cacheId ??= $this->cacheIdLoader->load();
 
         $criteria = new Criteria();
+        $criteria->addFilter(
+            new NotFilter(MultiFilter::CONNECTION_OR, [
+                new EqualsFilter('url', 'default.headless0')
+            ])
+        );
         $domains = $this->salesChannelDomainRepository->search($criteria, Context::createDefaultContext())->getEntities();
```

Comment:
Can you check that the salesChannelTypeId is storefront. so it's more generic :)